presto-main/src/test/java/com/facebook/presto/sql/planner/optimizations/TestMergeIdenticalWindows.java

```diff
@@ -0,0 +1,150 @@
+            PlanAssert.assertPlan(transactionSession, queryRunner.getMetadata(), actualPlan, pattern);
+            return null;
+        });
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsABA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "sum(quantity) over (partition by orderkey order by shipdate rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "quantity"), functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                        any()))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsABcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by orderkey order by shipdate rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount")),
+                                window(ImmutableList.of(functionCall("lag", "quantity", "*", "*")),
+                                        project(
+                                                window(ImmutableList.of(functionCall("sum", "quantity")),
+                                                        any()))))));
+    }
+
+    @Test
+    public void testIdenticalWindowSpecificationsAAcpA()
+    {
+        @Language("SQL") String sql = "select " +
+                "sum(quantity) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_quantity_A, " +
+                "lag(quantity, 1, 0.0) over (partition by suppkey order by orderkey rows between UNBOUNDED preceding and CURRENT ROW) sum_quantity_B, " +
+                "sum(discount) over (partition by suppkey order by orderkey rows between unbounded preceding and current row) sum_discount_A " +
+                "from lineitem";
+
+        assertUnitPlan(sql,
+                anyTree(
+                        window(ImmutableList.of(functionCall("sum", "discount"), functionCall("lag", "quantity", "*", "*")),
+                                project(
```

Comment:
How this `project` get in? You wrote in `MergeIdenticalWindows` that this optimizer works when all the windows are grouped in the plan (without any nodes inside).